lib/util/propTypes.js

```diff
@@ -151,22 +151,9 @@ module.exports = function propTypesInstructions(context, components, utils) {
       /** @type {UnionTypeDefinition} */
       const unionTypeDefinition = {
         type: 'union',
-        children: []
+        children: annotation.types.map(type => buildTypeAnnotationDeclarationTypes(type, parentName, seen))
       };
-      for (let i = 0, j = annotation.types.length; i < j; i++) {
-        const type = buildTypeAnnotationDeclarationTypes(annotation.types[i], parentName, seen);
-        // keep only complex type
```

Comment:
can you track down why this was added in the first place? it seems like something that might be important.

Reply:
`children === true` comes from b8e0a4ec85024c2d5860a723b0da377374ad6649 . But it does not help me understand why it is needed.

Reply:
cc @phpnode any insight here?

Reply:
no recollection, sorry :(